packages/squiggle-lang/src/rescript/ReducerInterface/ReducerInterface_GenericDistribution.res

```diff
@@ -66,6 +66,64 @@ module Helpers = {
       dist1,
     )->runGenericOperation
   }
+  let parseNumber = (args: expressionValue): Belt.Result.t<float, string> =>
+    switch args {
+    | EvNumber(x) => Ok(x)
+    | _ => Error("Not a number")
+    }
+
+  let parseNumberArray = (ags: array<expressionValue>): Belt.Result.t<array<float>, string> =>
+    E.A.fmap(parseNumber, ags) |> E.A.R.firstErrorOrOpen
+
+  let parseDist = (args: expressionValue): Belt.Result.t<GenericDist_Types.genericDist, string> =>
+    switch args {
+    | EvDistribution(x) => Ok(x)
+    | EvNumber(x) => Ok(GenericDist.fromFloat(x))
+    | _ => Error("Not a distribution")
+    }
+
+  let parseDistributionArray = (ags: array<expressionValue>): Belt.Result.t<
+    array<GenericDist_Types.genericDist>,
+    string,
+  > => E.A.fmap(parseDist, ags) |> E.A.R.firstErrorOrOpen
+
+  let mixtureWithGivenWeights = (
+    distributions: array<GenericDist_Types.genericDist>,
+    weights: array<float>,
+  ): DistributionOperation.outputType =>
+    E.A.length(distributions) == E.A.length(weights)
+      ? Mixture(Belt.Array.zip(distributions, weights))->runGenericOperation
```

Comment:
Reminder to self (and Quinn) that GenericOperation for Mixture does weighing, so we don't have to worry about that here.

Now that I think about it, it would probably have been good for that to have been an option in GenericOperation instead of here, but we can worry about that later.